src/modules/iotjs_module_websocket.c

```diff
@@ -244,8 +255,17 @@ JS_FUNCTION(PrepareHandshakeRequest) {
     return JS_CREATE_ERROR(COMMON, "Invalid host and/or path arguments!");
   };
 
-  generated_key = ws_generate_key(jsref);
-  size_t generated_key_len = strlen((char *)generated_key);
+  void *native_p;
+  JNativeInfoType *out_native_info;
+  bool has_p =
+      jerry_get_object_native_pointer(jsref, &native_p, &out_native_info);
+  if (!has_p || out_native_info != &wsclient_native_info) {
+    return WS_ERR_NATIVE_POINTER_ERR;
+  }
+  iotjs_wsclient_t *wsclient = (iotjs_wsclient_t *)native_p;
+
+  wsclient->generated_key = ws_generate_key(jsref);
+  size_t generated_key_len = 24;
```

Comment:
I'm curious on if we can assume the length of the key will constantly be 24.

Reply:
How about using `strnlen` if you can assume the maximum number of the key?

Reply:
Some reasons I think it's fine as it is:
1.: The `generated_key` ptr is not '\0' terminated, therefore if there are any valid characters behind it, it would get read by the function using the `generated_key_len`.
2.: The `ws_generate_key()` function always generates 16 random bytes, and base64 encodes them, which gives 24 constantly. The only exception is when the base64 fails, but in that case an exception is thrown, and execution is terminated.

Reply:
In case of the reason of 2, is the exception thrown for the base64 failure? Or else, just `error` event is emitted? If iot.js isn't immediately terminated at the exception, segment fault is able to occur because of `memcpy` from the source, `wsclient->generated_key`, NULL.

BTW, Is base64 encoding in `ws_generate_key` function able to be failed? All the inputs seem always valid.

Reply:
Hmm you are right, IoT.js should be terminated, since it is an uncaught exception, but the user could just catch the error, and continue running the program. 
The base64 encoding can fail as well.
By the way, thank you for clarifying your point, now I get it. You suggestion is right on using  `strlen`.

Reply:
@daeyeon @DanielBallaSZTE Thank you! I've updated the PR.